src/ImageSharp/Image/Image.LoadPixelData.cs

```diff
@@ -0,0 +1,100 @@
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <returns>A new <see cref="Image{TPixel}"/>.</returns>
+        public static Image<TPixel> LoadPixelData<TPixel>(TPixel[] data, int width, int height)
+            where TPixel : struct, IPixel<TPixel>
+            => LoadPixelData(Configuration.Default, data, width, height);
+
+        /// <summary>
+        /// Create a new instance of the <see cref="Image{TPixel}"/> class from the given pixel data.
+        /// </summary>
+        /// <param name="config">The config for the decoder.</param>
+        /// <param name="data">The byte array containing image data.</param>
+        /// <param name="width">The width of the final image.</param>
+        /// <param name="height">The height of the final image.</param>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <returns>A new <see cref="Image{TPixel}"/>.</returns>
+        public static Image<TPixel> LoadPixelData<TPixel>(Configuration config, TPixel[] data, int width, int height)
+            where TPixel : struct, IPixel<TPixel>
+            => LoadPixelData(config, new Span<TPixel>(data), width, height);
+
+        /// <summary>
+        /// Create a new instance of the <see cref="Image{TPixel}"/> class from the given byte array as raw pixel data.
+        /// </summary>
+        /// <param name="data">The byte array containing image data.</param>
+        /// <param name="width">The width of the final image.</param>
+        /// <param name="height">The height of the final image.</param>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <returns>A new <see cref="Image{TPixel}"/>.</returns>
+        public static Image<TPixel> LoadPixelData<TPixel>(byte[] data, int width, int height)
+            where TPixel : struct, IPixel<TPixel>
+            => LoadPixelData<TPixel>(Configuration.Default, data, width, height);
+
+        /// <summary>
+        /// Create a new instance of the <see cref="Image{TPixel}"/> class from the given byte array as raw pixel data.
+        /// </summary>
+        /// <param name="config">The config for the decoder.</param>
+        /// <param name="data">The byte array containing image data.</param>
+        /// <param name="width">The width of the final image.</param>
+        /// <param name="height">The height of the final image.</param>
+        /// <typeparam name="TPixel">The pixel format.</typeparam>
+        /// <returns>A new <see cref="Image{TPixel}"/>.</returns>
+        public static Image<TPixel> LoadPixelData<TPixel>(Configuration config, byte[] data, int width, int height)
+            where TPixel : struct, IPixel<TPixel>
+        {
+            int size = width * height;
+            using (var sourceBuffer = new Buffer<TPixel>(size))
+            {
+                PixelOperations<TPixel>.Instance.PackFromRawBytes(new Span<byte>(data), sourceBuffer.Span, size);
```

Comment:
What was your goal with `LoadPixelData<TPixel>(Configuration config, byte[] data, int width, int height)` ? Is it meant to be a stream in `TPixel` format or is it meant to be a stream of `Rgba32`.

- It's not well documented, the user needs to guess.
- If the `sourceBytes` meant to have it's memory layout in `TPixel` format, we don't need `PackFromRawBytes()`. It's always a memcpy, you can just call `SpanHelpers.Copy(data, buffer.Span.AsBytes(), size)`
- If `sourceBytes` meant to be in `Rgba32`, your `PackFromRawBytes()` implementation is wrong!

Personally I think we need the first variant here: loading byte data having identical memory layout to `TPixel[]`. In this case you just need to unsafe cast your `byte` span to `TPixel` span, no need for an extra copy.